Remove a bunch of old elastic beanstalk setup code

ttung opened the pull request:
### Description
These configuration profiles and documentation pertain to the elastic beanstalk steup.

Depends on #346 

### Test plan
GHA

phoenixAja commented:
lol i was just thinking of doing this

jgadling commented on src/backend/scripts/setup_dev_data.sh:
```diff
@@ -29,5 +29,9 @@ ${local_aws} secretsmanager update-secret --secret-id aspen-config --secret-stri
-  "AUTH0_CLIENT_KWARGS": {"scope": "openid profile email offline_access"}
+  "AUTH0_CLIENT_KWARGS": {"scope": "openid profile email offline_access"},
+  "FLASK_SECRET": "DevelopmentKey",
+  "DB.rw_username": "user_rw",
```
Can we not put "." chars in keys? It's pretty confusing

ttung replied:
Updated to use `/` as well.

phoenixAja commented on src/backend/aspen/cli/db.py:
```diff
@@ -38,9 +36,8 @@
-@click.option("--remote", "config_cls", flag_value=RemoteDatabaseConfig)
-@click.option("--docker", "config_cls", flag_value=LocalConfig)
+@click.option("--remote", "config_cls", flag_value=Config)
+@click.option("--docker", "config_cls", flag_value=DockerComposeConfig, default=True)
```
i like `local` as a flag here instead of `docker`, but that's a small nit

ttung replied:
I think of `--local` as without docker, but I can change it.

phoenixAja reviewed (approved):
this looks good to me, do we have to change the docs anywhere to reflect these changes?

Commit:
Remove a bunch of old elastic beanstalk setup code

### Description
These configuration profiles and documentation pertain to the elastic beanstalk steup.

### Test plan
GHA
